src/core/locator/qgslocatorwidgetcore.h

```diff
@@ -0,0 +1,97 @@
+/***************************************************************************
+                         qgslocatorwidgetcore.h
+                         ------------------
+    begin                : November 2018
+    copyright            : (C) 2018 by Denis Rouzaud
+    email                : denis@opengis.ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSLOCATORWIDGETCORE_H
+#define QGSLOCATORWIDGETCORE_H
+
+#include <QObject>
+
+#include "qgis_core.h"
+
+class QgsLocatorResult;
+class QgsLocator;
+class QgsLocatorContext;
+class QgsLocatorModel;
+class QgsLocatorProxyModel;
+class QgsMapCanvasInterface;
+
+
+/**
+ * \ingroup core
+ * The QgsLocatorWidgetCore class provides the core functionality
+ * to be used in a locator widget.
+ * \since QGIS 3.6
+ */
+class CORE_EXPORT QgsLocatorWidgetCore : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY( bool isRunning READ isRunning NOTIFY isRunningChanged )
+  public:
+    //! Constructor of QgsLocatorWidgetCore
+    explicit QgsLocatorWidgetCore( QObject *parent = nullptr );
+
+    //! Set the map canvas interface
+    void setMapCanvasInterface( QgsMapCanvasInterface *canvasInterface );
+
+    //! Perform a search
```

Comment:
Better dox here -- is this blocking? If not, what happens to existing searches? Does it block until they are canceled?